Created functionality for saving unknown telegram users

ChernenkoVitaliy opened the pull request:
# GreenCity UBS PR

## Summary Of Changes 🔥

- Added entity `UnknownTelegramUser`;
- Added `UnknownTelegramUserRepository`.java;
- Added `changelog ch-create-table-unknown-telegram-user-Chernenko.xml` for creating table for unknown telegram users;
- Included created changelog into `db.changelog-master.xml`;
- Changed returning type from `TelegramBoto` to Optional for findByChatId method in `TelegramBotRepository`.java;
- Added a few constants into `AppConstant`.java for telegram bot;
- Refactored method _onUpdateReceived_ in `UBSTelegramBot` for handling unknown telegram users;
- Added `MessageFactory` for creating `SendMessage` for tg bot;
- Added `TelegramExecutor` for separete execute method and Exception handling for it from main business logic;
- Added unit tests for method _onUpdateReceived_ in `UBSTelegramBotTest`;
- Deleted `TelegramBotAlreadyConnected`.java;
- Renamed old `TelegramService` to `TelegramNotificationService`;
- Added a new `TelegramService` interface and `TelegramServiceImpl` as realization of it;
- Created unit tests for TelegramServiceImpl;
- Added telegram dependency into `pom.xml` in a _service-api_ module;
- Moved constant for telegram version in main `pom.xml`;

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Added support for handling unknown Telegram users
  - Implemented Telegram bot integration with user management

- **Improvements**
  - Enhanced Telegram bot interaction logic
  - Improved error handling for Telegram user registration
  - Updated Telegram bot service to manage user connections more efficiently

- **Technical Updates**
  - Updated Telegram bot library to version 5.4.0.1
  - Restructured Telegram service implementation
  - Added new repositories and entities for Telegram user tracking

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ChernenkoVitaliy committed:
Added entity UnknownTelegramUser;
Added UnknownTelegramUserRepository.java;
Added changelog ch-create-table-unknown-telegram-user-Chernenko.xml for creating table for unknown telegram users;
Included created changelog into db.changelog-master.xml;
Changed returning type from TelegramBoto to Optional for findByChatId method in TelegramBotRepository.java;
Added few constants into AppConstant.java for telegram bot;
Refactored method onUpdateReceived in UBSTelegramBot for handling unknown telegram users;

ChernenkoVitaliy committed:
Deleted TelegramBotAlreadyConnected.java;
Added MessageFactory for creating SendMessage for tg bot;
Added TelegramExecutor for separete execute method and Exeption handling for it from main bussines logic;
Replaced execute method call in UBSTelegramBot by using TelegramExecutor and MessageFactory;
Added unit tests for method onUpdateReceived in UBSTelegramBotTest;

ChernenkoVitaliy committed:
Deleted TelegramBotAlreadyConnected.java;
Added MessageFactory for creating SendMessage for tg bot;
Added TelegramExecutor for separete execute method and Exeption handling for it from main bussines logic;
Replaced execute method call in UBSTelegramBot by using TelegramExecutor and MessageFactory;
Added unit tests for method onUpdateReceived in UBSTelegramBotTest;

ChernenkoVitaliy committed:
Formatted code using formatter-maven-plugin;
Refactored unit tests for UBSTelegramBotTest;
Replased string to constatn from AppConstant class in MessageFactory;

Cr1stal423 commented on service/src/main/java/greencity/ubstelegrambot/UBSTelegramBot.java:
```diff
@@ -41,23 +43,52 @@ public String getBotToken() {
+                    .id(update.getMessage().getFrom().getId())
+                    .firstName(update.getMessage().getFrom().getFirstName())
+                    .lastName(update.getMessage().getFrom().getLastName())
+                    .userName(update.getMessage().getFrom().getUserName())
```
It may be better to move this builder to another class

Cr1stal423 commented on service/src/main/java/greencity/ubstelegrambot/UBSTelegramBot.java:
```diff
@@ -41,23 +43,52 @@ public String getBotToken() {
+                unknownTelegramUserRepository.delete(unknownSavedTelegramUser.get());
+                telegramBotRepository.save(getTelegramBot(user, tgUserId));
+                executor.executeCommand(this, MessageFactory.creatWelcomeMessage(tgUserId.toString()));
+                return;
```
The situation is the same. Is it possible to put this code in the service?

VitaliiChernenko replied:
The whole business logic from UBSTelegramBot was moved into TelegramService.
Thanks, it was a good point.

Cr1stal423 commented on dao/src/main/java/greencity/entity/telegram/UnknownTelegramUser.java:
```diff
@@ -0,0 +1,29 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@EqualsAndHashCode
```
@EqualsAndHashCode is already in the @Data annotation

ChernenkoVitaliy committed:
Renamed old TelegramService to TelegramNotificationService;
Renamed old  TelegramServiceTest to TelegramNotificationServiceTest;
Added telegrambots-spring-boot-starter dependency into pom.xml in service-api module;
Moved telegrambot.version constant to main pom.xml;
Added java doc to TelegramExecutor, MessageFactory, TelegramService;
Added TelegramService interface and TelegramServiceImpl class;
Added unit tests for TelegramServiceImplTest and UBSTelegramBotTest;
Refactored onUpdateReceived method in UBSTelegramBot class;

ChernenkoVitaliy committed:
Fixed name of method in MessageFactory;
Removed telegram dependency from pom.xml in service module cause it has been added in pom.xml in service-api module;
